barretenberg/cpp/src/barretenberg/common/serialize.hpp

```diff
@@ -371,7 +371,8 @@ template <typename B, typename T> inline void read(B& it, std::optional<T>& opt_
 }
 
 template <typename T>
-concept HasGetAll = requires(T t) { t.get_all(); };
+concept HasGetAll = requires(T t) { t.get_all(); } && !
+msgpack_concepts::HasMsgPack<T>;
```

Comment:
added because otherwise, VerificationKey satisfies both HasGetAll and HasMsgPack, which leads to an ambiguous call error.